src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -300,6 +284,56 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 								 relationShardList, QTW_EXAMINE_RTES_BEFORE);
 	}
 
+	if (IsA(node, OnConflictExpr))
+	{
+		OnConflictExpr *onConflict = (OnConflictExpr *) node;
+		Oid constraintId = onConflict->constraint;
+
+		if (constraintId != InvalidOid)
+		{
+			Oid constraintRelationId = InvalidOid;
+
+			HeapTuple tp = SearchSysCache1(CONSTROID, ObjectIdGetDatum(constraintId));
+			if (HeapTupleIsValid(tp))
+			{
+				Form_pg_constraint contup = (Form_pg_constraint) GETSTRUCT(tp);
+
+				constraintRelationId = contup->conrelid;
+				ReleaseSysCache(tp);
+			}
+
+			/*
+			 * we can return here without calling the walker function, since we know there will be no
+			 * possible tables or constraints after this point, by the syntax.
+			 */
+			if (constraintRelationId == InvalidOid)
+			{
+				return false;
```

Comment:
Is it possible except buggy cases, should we throw an error instead ?

Reply:
or, might logging error be an option? @onderkalaci what do you think?

Reply:
Well, as you noted @onurctirtir I cannot see how `if (!OidIsValid(constraintRelationId))` returns true as we have `if (OidIsValid(constraintId))`. Still, might be fine to throw an error:

So, something like the following might work
```C
if (!OidIsValid(constraintRelationId))
{
     ereport(..., "unexpected relation id (%u) for constraint: %s", constraintRelationId, get_constraint_name(..))
}
```